unit-tests/run-unit-tests.py

```diff
@@ -90,8 +123,35 @@ def debug(*args):
     elif opt in ('-q','--quiet'):
         def out(*args):
             pass
-if len(args) != 1:
+    elif opt in ('-r', '--regex'):
+        regex = args[0]
+        del args[0]
+if len(args) > 1:
     usage()
+target = None
+if len(args) == 1:
+    if os.path.isdir( args[0] ):
+        target = args[0]
+    else:
+        usage()
+# Trying to assume target directory from inside build directory. Only works if there is only one location with tests
+if not target:
+    build = librealsense + os.sep + 'build'
+    for executable in find(build, '(^|/)test-.*'):
+        if not is_executable(executable):
+            continue
+        dir_with_test = build + os.sep + os.path.dirname(executable)
+        if target and target != dir_with_test:
```

Comment:
When is this ever true? Isn't `target` always None here? (line 138)

Reply:
its true when there are 2 directories with test executables

Reply:
Oh I see! I missed line 146